ext/opentelemetry-ext-azure-monitor/src/opentelemetry/ext/azure_monitor/metrics.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+import requests
+
+from opentelemetry.ext.azure_monitor import protocol, util
+from opentelemetry.sdk.metrics.export import (
+    MetricsExporter,
+    MetricsExportResult,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMonitorMetricsExporter(MetricsExporter):
+    def __init__(self, **options):
+        self.options = util.Options(**options)
+        if not self.options.instrumentation_key:
+            raise ValueError("The instrumentation_key is not provided.")
+
+    def export(self, metric_tuples):
+        # Metric tuples is a sequence of metric to label values pairs
+        envelopes = tuple(map(self.metric_tuple_to_envelope, metric_tuples))
+
+        try:
```

Comment:
This piece of code is >95% duplicated with https://github.com/open-telemetry/opentelemetry-python/blob/250b9faed8d5ca4741dd9b8ede3aff7947c70167/ext/opentelemetry-ext-azure-monitor/src/opentelemetry/ext/azure_monitor/trace.py#L38.

Reply:
I moved this logic to a separate class `TransportMixin`.